lib/corpus.py

```diff
@@ -301,6 +303,172 @@ def _dump_internal(self, out: TextIO, indentation: str = ""):
             out.write("%s</subcorpus>\n" % (indentation,))
 
 
+class CorpusV2(Corpus):
+    """
+    This class represents a corpus in the Bliss format. It is also used to represent subcorpora when the parent_corpus
+    attribute is set. Corpora with include statements can be read but are written back as a single file.
+
+    The difference with respect to :class:`i6_core.lib.corpus.Corpus` is that in this class we can access
+    any recording or segment in practically O(1).
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        self.parent_corpus: Optional[Corpus] = None
+
+        self.subcorpora: Dict[str, Corpus] = {}
+        self.recordings: Dict[str, RecordingV2] = {}
```

Comment:
You must not redefine the types of attribs in a subclass in an incompatible way. (The only thing which is allowed is to specify/restrict them more, e.g. changing from `Dict[str, Any]` to `Dict[str, int]` in a subclass is ok.)

Reply:
Then I shall not inherit from `Corpus`.

Reply:
No, that's not necessarily true. You could also just rename them. And then maybe define `subcorpora`/`recordings` as property.

Reply:
That's basically also what @michelwi said.